### PR TITLE
Fix and enable few ORTModule Unit Tests

### PR DESCRIPTION
**Pull request by pengwa**
### Fix and enable few ORTModule Unit Tests

Fix 'test_bert_inputs_with_dynamic_shape' and 'test_bert_result_with_layerwise_recompute' generate Nan loss in ORT run. 

The root cause is, the logic to generatic attention mask test data is not correct, only 0 or 1 is allowed in the dataset, but we see lots of other numbers. ( The reason we don't have this using old version of transformers for example v4.4.2 or 4.16.2 is because they don't contains such https://github.com/huggingface/transformers/commit/d3cb28886ac68beba9a6646b422a4d727b056c0c, which increase the scaling to a bigger number, causing a overflow to inf) 

Another improvement during the investigation using convergence tools:
Don't dump the activations during model export phase, otherwise, the dumped data might contains some PyTorch run's result making us confused during comparing with stock PyTorch run results.

### CODE REVIEW

**Review by baijumeswani — APPROVED** [review]

**Comment by pengwa** [comment]
Thanks @baijumeswani !
